front/app/modules/commercial/id_vienna_saml/components/ViennaSamlButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { AUTH_PATH } from 'containers/App/constants';
+
+// i18n
+import messages from './messages';
+import { FormattedMessage } from 'utils/cl-intl';
+
+import { StyledAuthProviderButton } from 'components/SignUpIn/AuthProviders';
+import { TOnContinueFunction } from 'components/SignUpIn/AuthProviderButton';
+
+// typings
+import { TSignUpInFlow } from 'components/SignUpIn';
+
+interface Props {
+  flow: TSignUpInFlow;
+  onContinue: TOnContinueFunction;
+}
+
+const ViennaSamlButton = ({ onContinue, flow }: Props) => {
+  const setHref = () => {
+    window.location.href = `${AUTH_PATH}/vienna_citizen`;
+  };
+  const handleOnContinue = () => {
+    if (flow === 'signup') {
+      window.location.replace(
+        'https://mein.wien.gv.at/Registrieren?branding=citizenlab'
+      );
+    } else {
+      onContinue('id_vienna_saml', setHref);
+    }
+  };
+  return (
+    <StyledAuthProviderButton
+      authProvider="id_vienna_saml"
+      onContinue={handleOnContinue}
+      flow={flow}
+      showConsentOnFlows={['signin']}
```

Comment:
I would indeed either change it to a `string` (like Alex says) or define this array not in-line but as a `const` outside of the component. Now that it's wrapped in an array, the `useEffect` inside of `front/app/components/SignUpIn/AuthProviderButton.tsx` is going to fire on every re-render because arrays are only compared by reference, and like this you get a fresh new array on every re-render.

Reply:
The idea of making this an array is that you could have an AuthButton that shows the consent on ['signup', 'signin'], so both flows. I can change it to be just a simple string though, then only one of the flows can have consent message shown.